### PR TITLE
use given locale in translate

### PR DESCRIPTION
**Pull request by mameier**
Temporarily change the used locale in translate.

FastGettext.locale was not set to the locale parameter.
If locale was changed by parameter, the cache of the thread locale was always returned.

This caused route_translator to create only routes for the default locale.

### CODE REVIEW

File: spec/gettext_i18n_rails/backend_spec.rb [hunk 1]
```diff
@@ -77,16 +78,28 @@
       subject.translate('xx', 'c', {}).should == 'd'
     end
```
Comment (mameier) [inline spec/gettext_i18n_rails/backend_spec.rb]: new test case, failing without the patch

File: spec/gettext_i18n_rails/backend_spec.rb [hunk 2]
```diff
@@ -77,16 +78,28 @@
...
 
+    it 'temporarily sets the given locale' do
```
Comment (mameier) [inline spec/gettext_i18n_rails/backend_spec.rb]: new test case, failing without the patch

**Comment by grosser** [comment]
is there a `FastGettext.with_locale` helper ? ... if not, can you add/use that ?

File: spec/gettext_i18n_rails/backend_spec.rb [hunk 3]
```diff
@@ -77,16 +78,28 @@
...
+      FastGettext.should_receive(:set_locale).with('xx').and_return('yz')
+      FastGettext.should_receive(:set_locale).twice.with('yz').and_return('xx')
+      subject.backend.should_receive(:translate).with('xx', 'c', {}).and_return 'd'
+      FastGettext.locale= 'yz'
+      FastGettext.stub(:current_repository).and_return 'a'=>'b'
+      subject.translate('xx', 'c', {}).should == 'd'
+
```
Comment (grosser) [inline spec/gettext_i18n_rails/backend_spec.rb]: extra whitespace

File: spec/gettext_i18n_rails/backend_spec.rb [hunk 4]
```diff
@@ -77,16 +78,28 @@
       subject.translate('xx', 'c', {}).should == 'd'
     end
 
+    it 'temporarily sets the given locale' do
+      FastGettext.should_receive(:set_locale).with('xx').and_return('yz')
+      FastGettext.should_receive(:set_locale).twice.with('yz').and_return('xx')
+      subject.backend.should_receive(:translate).with('xx', 'c', {}).and_return 'd'
+      FastGettext.locale= 'yz'
```
Comment (grosser) [inline spec/gettext_i18n_rails/backend_spec.rb]: prefer the extra space `FastGettext.locale = 'xy'`

File: spec/gettext_i18n_rails/backend_spec.rb [hunk 5]
```diff
@@ -77,16 +78,28 @@
       subject.translate('xx', 'c', {}).should == 'd'
     end
 
+    it 'temporarily sets the given locale' do
+      FastGettext.should_receive(:set_locale).with('xx').and_return('yz')
```
Comment (grosser) [inline spec/gettext_i18n_rails/backend_spec.rb]: not a big fan of all the stubbing ... can the test only assert that the backend was called with the right locale and the locale is whatever it was before calling instead ?

**Comment by grosser** [comment]
looks about right

**Comment by grosser** [comment]
aaahh wrong button :D

File: spec/gettext_i18n_rails/backend_spec.rb (hunk 5 above)
Comment (mameier) [inline spec/gettext_i18n_rails/backend_spec.rb]: I don't like the stubbing ether, but if I leave it out, the real caching is performed and all the other tests won't call :current_repository any longer. So this stubbing isolates the tests.

**Comment by mameier** [comment]
unfortunately FastGettext does not have .with_locale
If I add and use it, it would produce a dependency to a not released gem.
So I could make a pull request on FastGettext and we could clean up gettext_i18n_rails when it's released.

**Comment by grosser** [comment]
You can make the PR or I'll make it tomorrow morning, it's a nice feature
on it's own, so I'll just release that.

**Comment by mameier** [comment]
Hi Michael,

I have an new minor pull request: Fix compatibility with HAML 6.
HAML 6 change the api of "Engine".

That change now adds to the pull request of 2016.

Is there a reason, why you didn't release my old changes, as you announced?

**Comment by grosser** [comment]
no idea :D
... but here comes 1.9.0 :)

**Pull request merged**